migrations/2_deploy.js

```diff
@@ -31,19 +28,19 @@ module.exports = async function(deployer, network, accounts) {
   // Creating DeployerApp helper.
   const deployerApp = new DeployerApp(deployer, web3, deployerAccount, network);
 
-  await deployerApp.deploys([ AddressLib, SafeMath, ZDai ], deployOptions);
```

Comment:
Why did you remove the libraries?

Reply:
You only need to deploy and link a library if it has `external` functions that you use. If all of the functions are internal, then they are copied into your contract upon compilation. You therefore do not need to deploy or link the library.